### PR TITLE
feat(helm/teleport-kube-agent): custom annotations in the Secret

### PR DESCRIPTION
**Pull request by danmx**
Adding support for custom annodations in the kube agent Secret.
This makes it easier to work with e.g. [ArgoCD Vault Plugin](https://argocd-vault-plugin.readthedocs.io/en/stable/howitworks/).

Signed-off-by: danmx <daniel.iziourov@elastic.co>

### CODE REVIEW

**Comment by danmx** [comment]
I've added docs changes later so this PR is missing `documentation` label.

**Review by tigrato — APPROVED** [review]

File: examples/chart/teleport-kube-agent/values.yaml [hunk 1]
```diff
@@ -354,6 +354,8 @@ annotations:
   deployment: {}
   # Annotations for each Pod in the Deployment
   pod: {}
+  # Annotations for the Secret (ignored when disabled)
```
Comment (hugoShaka) [inline examples/chart/teleport-kube-agent/values.yaml:357]:
```suggestion
  # Annotations for the Secret (has no effect when `joinTokenSecret.create` is false)
```

**Review by hugoShaka — APPROVED** [review]

File: examples/chart/teleport-kube-agent/values.yaml (hunk 1 above)
Comment (danmx) [reply examples/chart/teleport-kube-agent/values.yaml]: fixed though this comment was taken from `extraLabels.secret`

**Comment by hugoShaka** [comment]
@danmx thank you for your contribution 🎉 

Our current CI pipeline doesn't like external contributions, I created an internal PR with your commits here: https://github.com/gravitational/teleport/pull/30761, you will retain authorship.

I'll close this PR in favor of [the internal one](https://github.com/gravitational/teleport/pull/30761).

**Pull request closed without merging**